src/transformers/models/aria/modular_aria.py

```diff
@@ -0,0 +1,1357 @@
+            resample (PILImageResampling, optional): The resampling filter to use if resizing the image. Defaults to BICUBIC.
+
+        Returns:
+            BatchFeature: A BatchFeature object containing:
+                - 'pixel_values': Tensor of processed image pixel values.
+                - 'pixel_mask': Boolean pixel mask. This mask is a 2D tensor of shape (max_size, max_size) where:
+                    - True (1) values indicate pixels that belong to the original resized image.
+                    - False (0) values indicate pixels that are part of the padding.
+                  The mask helps distinguish between actual image content and padded areas in subsequent processing steps.
+                - 'num_crops': The maximum number of crops across all images.
+        """
+        max_size = self.max_image_size if max_image_size is None else max_image_size
+        min_size = self.min_image_size if min_image_size is None else min_image_size
+
+        if max_size not in [490, 980]:
+            raise ValueError("max_image_size must be either 490 or 980")
+
+        if not isinstance(images, list):
+            images = [images]
+
+        pixel_values = []
+        pixel_masks = []
+        num_crops = None
+
+        for image in images:
+            if do_convert_rgb:
+                image = convert_to_rgb(image)
+            image = to_numpy_array(image)
+            if split_image:
+                crop_images = self.get_image_patches(image, self.split_ratio, max_size)
+            else:
+                crop_images = [image]
+            if num_crops is None or len(crop_images) > num_crops:
+                num_crops = len(crop_images)
+            for crop_image in crop_images:
+                # At this point the scale is the rescaling factor that would bring the image to max_size in its larger dimension
+                h, w = crop_image.shape[:2]
+                scale = max_size / max(h, w)
+                if w >= h:
+                    new_size = (max(int(h * scale), min_size), max_size)  # h, w
+                else:
+                    new_size = (max_size, max(int(w * scale), min_size))  # h, w
+
+                crop_image_resized = resize(crop_image, new_size, resample=resample)
+
+                padding_bottom, padding_right = max_size - new_size[0], max_size - new_size[1]
+                crop_image_padded = pad(crop_image_resized, ((0, padding_bottom), (0, padding_right)))
+
+                # Create a pixel mask
+                pixel_mask = torch.zeros(max_size, max_size, dtype=bool)
+                pixel_mask[: new_size[0], : new_size[1]] = 1
+                pixel_masks.append(pixel_mask)
+
+                if do_normalize:
+                    crop_image_padded = self.normalize(crop_image_padded, self.image_mean, self.image_std)
+
+                # Switch to rgb channel first
+                crop_image_padded = np.transpose(crop_image_padded, (2, 0, 1))
```

Comment:
channel should be defined with `data_format` kwarg and `to_channel_dimension_format` helper from image utils, so users can specify any channel format